ansible/roles/aws_repo_setup/tasks/main.yml

```diff
@@ -1,16 +1,52 @@
 ---
-  - name: install EPEL
```

Comment:
If I'm reading this right, you removed EPEL.

You don't want to remove it, you want to support multiple ways of using content:
 - EPEL
 - RedHat CDN
 - Mirror Repos

Might be each is written in it's own task file, then you include the one you want based on some variables.